spec/features/accept_language_spec.rb

```diff
@@ -0,0 +1,9 @@
+describe "extract locale from client when user has a browser setup in en-US" do
+  
+  let(:accept_language){ "en-US,en;q=0.8,es-419;q=0.6,es;q=0.4,ca;q=0.2,af;q=0.2" }
+    
+  it "parses Accept-Language header into locale string en" do
```

Comment:
This is a good start, but remember the golden rule: tests must be **correct** and **complete**.

This is clearly not complete, it only covers a single string. Does it work for `ar` too?

Can you gather a few extra real examples and add them here?